pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -186,7 +187,7 @@ func validateImmutableFieldsCluster(new, old *Cluster) field.ErrorList {
 		if !oldAWSIamConfig.Equal(newAWSIamConfig) {
 			allErrs = append(
 				allErrs,
-				field.Forbidden(field.NewPath("spec", "AWS Iam Config"), fmt.Sprintf("field is immutable %v", newAWSIamConfig.Kind)))
+				field.Forbidden(specPath.Child("AWS Iam Config"), fmt.Sprintf("field is immutable %v", newAWSIamConfig.Kind)))
```

Comment:
This is not a field right? It's the entire config, unless this is referring to the ref field, which we should refer to that instead.

Reply:
Not a field technically, its a ref. But when users edit it, we treat it as a field that is a reference.

Reply:
```suggestion
				field.Forbidden(specPath.Child("identityProviderRefs", "kind", "AWSIamConfig"), fmt.Sprintf("field is immutable %v", newAWSIamConfig.Kind)))
```